### PR TITLE
feat(app): Add release notes to robot update modals

### LINKED ISSUE #2353: Robot update: Display release notes
## overview

As a user, I would like to know what changes I'm about to push to my robot's software when pushing an update.

## current behavior

- Robot update modal shows generic message about keeping your robot up to date

## acceptance criteria 

- [x] Robot update modal shows "API" section of the release notes of the update that is about to be applied

## implementation details

Current release notes are present in the app UI already:

```js
import {CURRENT_RELEASE_NOTES} from './path/app/src/shell'
```

`CURRENT_RELEASE_NOTES` is a plain string, and the "API" section is in between two comment nodes:

```md
<!-- start:@opentrons/api -->
## OT2 and Protocol API

Blah blah blah

...
<!-- end:@opentrons/api -->
```

The robot update modal should be aligned with the app update modal and pass the snipped out portion of the release notes (using the comment nodes as anchors) to the `ReleaseNotes` markdown-rendering component.

This ticket takes over the robot update portion of and closes #2028

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR closes #2353 with a refactor to move `ReleaseNotes` and `UpdateModal` out of `AppSettings` and then adding in API only release notes to the RobotUpdateModal.

screen shot 2018-10-01 at 4 01 39 pm

## changelog

- feat(app): Add release notes to robot update modals

## review requests

Change `package.json` app version to something earlier like `3.3.0`
- [ ] AppUpdateModal still renders as expected
- [ ] RobotUpdateModal shows release notes (API only)
- [ ] RobotUpdateModal shows update app message followed by downgrade robot message before release notes

*note: I removed app release notes rather than extracting api release notes. This way we still have the changes heading and link to technical release notes *

### CODE REVIEW

File: app/src/components/ReleaseNotes/styles.css [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  & li {
+    margin: 0.25rem 0;
+  }
+
+  & code {
+    font-family: monospace;
+    color: var(--c-font-light);
+  }
+
+  & li > code {
```
Comment (mcous) [inline app/src/components/ReleaseNotes/styles.css]: Good catch on this bug. Can we make the default `color` for `code` dark and then only make it light if it's inside a `pre`? I think it's a little easier to reason with that way

```css
& code {
  font-family: monospace;
  color: var(--c-font-dark);
}

& pre {
  margin: 0.5rem 0;
  padding: 0.5rem 0.75rem;
  background-color: var(--c-font-dark);
    
  & code {
    color: var(--c-font-light);
  }
}
```

File: app/src/components/RobotSettings/UpdateRobotMessage.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// @flow
+import * as React from 'react'
+import styles from './styles.css'
+
+type Props = {message: React.Node}
+
+export default function UpdateAppMessage (props: Props) {
```
Comment (mcous) [inline app/src/components/RobotSettings/UpdateRobotMessage.js]: Copy paste typo with component name

File: app/src/components/AppSettings/AppUpdateModal.js [hunk 3]
```diff
@@ -2,11 +2,10 @@
 // AlertModal for updating to newest app version
 import * as React from 'react'
 
-import {Icon, AlertModal, type ButtonProps} from '@opentrons/components'
-import {Portal} from '../portal'
-import {BottomButtonBar} from '../modals'
-import ReleaseNotes from './ReleaseNotes'
-import styles from './styles.css'
+import {Icon, type ButtonProps} from '@opentrons/components'
+// import {Portal} from '../portal'
```
Comment (mcous) [inline app/src/components/AppSettings/AppUpdateModal.js]: Did you mean to delete this?

File: components/src/modals/modals.css [hunk 4]
```diff
@@ -112,6 +112,7 @@
 
   max-height: 100%;
   padding: 1rem;
+  overflow-y: auto;
```
Comment (mcous) [inline components/src/modals/modals.css]: Is this addition necessary? I think the `overflow-y` you added to `.update_modal_scroll` isn't actually working because `.update_modal_scroll` needs a defined height or max-height, so this is just covering that bug

File: app/src/components/modals/styles.css [hunk 5]
```diff
@@ -38,3 +38,15 @@
...
+.update_modal {
+  max-height: 100%;
+}
+
+.update_modal_contents {
+  display: flex;
+}
+
+.update_modal_scroll {
+  overflow-y: auto;
```
Comment (mcous) [inline app/src/components/modals/styles.css]: I don't think this will do anything unless you also add a `max-height: 100%` (see comment in `components/src/modals/modals.css`)

File: app/src/components/ReleaseNotes/styles.css (hunk 1 above)
Comment (Kadee80) [reply app/src/components/ReleaseNotes/styles.css]: 👍

File: app/src/components/AppSettings/AppUpdateModal.js (hunk 3 above)
Comment (Kadee80) [reply app/src/components/AppSettings/AppUpdateModal.js]: Good catch.

File: app/src/components/modals/styles.css (hunk 5 above)
Comment (Kadee80) [reply app/src/components/modals/styles.css]: Going to add in max height here and leave the AalertModal styles in components as they were before.

File: components/src/modals/modals.css (hunk 4 above)
Comment (Kadee80) [reply components/src/modals/modals.css]: removing in favor of app specific css
